decidim-core/app/controllers/decidim/user_timeline_controller.rb

```diff
@@ -12,7 +12,7 @@ class UserTimelineController < Decidim::ApplicationController
     helper_method :activities, :resource_types, :user
 
     def index
-      raise ActionController::RoutingError, "Not Found" if current_user != user
+      raise ActionController::RoutingError, "Not Found" if !user || current_user != user
```

Comment:
It is a bit strange to have it like "if no user or current user is different than user"
```suggestion
      raise ActionController::RoutingError, "Not Found" unless user && current_user == user
```